src/App.tsx

```diff
@@ -15,36 +15,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  alphabet = 'alphabet',
+  length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>(goodsFromServer);
+  const [isActive, setIsActive] = useState<SortField>();
+  const [reverse, setReverse] = useState(false);
+
+  const resetSort = () => {
+    setGoods([...goodsFromServer]);
+    setIsActive(undefined);
+    setReverse(false);
+  };
+
+  const reversGoods = (foods: string[]) => {
+    setReverse(!reverse);
+    setGoods([...foods].reverse());
+  };
+
+  const sortedGoods = (type: SortField): void => {
+    setGoods(
+      [...goods].sort((good1, good2) => {
```

Comment:
Create sort function in some helper file. This function should retreive 3 arguments: array of goods, sort type, and reverse values. The function have to sort the array and reverse it if third argument equals true. Also for such purposes it's better to use switch case construction. Please feel free to ask for help in fe_chat.